api/src/test/java/org/apache/iceberg/variants/TestSerializedPrimitives.java

```diff
@@ -182,10 +182,10 @@ public void testNegativeDouble() {
   public void testDecimal4() {
     VariantPrimitive<?> value =
         SerializedPrimitive.from(
-            new byte[] {primitiveHeader(8), 0x04, (byte) 0xD2, 0x02, (byte) 0x96, 0x49});
+            new byte[] {primitiveHeader(8), 0x04, (byte) 0x15, (byte) 0xCD, (byte) 0x5B, 0x07});
 
     assertThat(value.type()).isEqualTo(PhysicalType.DECIMAL4);
-    assertThat(value.get()).isEqualTo(new BigDecimal("123456.7890"));
```

Comment:
So is the issue here that 123456.7890 == Precision 10 so by the spec it shouldn't be Decimal4 it should be Decimal 8

Reply:
That's correct. Other engines like Spark and Parquet-Java implements properly.